File: java/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/Elements/ConcreteElementA.java
package Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements;

import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Visitors.Visitor;

public class ConcreteElementA implements Element {
    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    public String exclusiveMethodOfConcreteElementA() {
        return "A";
    }
}

File: typescript/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementA.ts
import {Element} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/Element";
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/Visitor";

class ConcreteElementA implements Element {
    public accept(visitor: Visitor): void {
        visitor.visitConcreteElementA(this);
    }
    public exclusiveMethodOfConcreteElementA(): string {
        return 'A';
    }
}

export { ConcreteElementA };